### ISSUE TITLE
Cloud to Device messages are not reaching client/device.

### ISSUE THREAD

**Issue by bmrpatel** [Azure/azure-iot-sdk-node#40]
# Context
- **OS and version used:** Ubuntu 16.04
- **Node.js version:** 7.5
- **npm version:** 4.1.2
- **list of installed packages:** 
"azure-iot-device": "^1.1.11",
    "azure-iot-device-amqp": "^1.1.11",
    "azure-iot-device-mqtt": "^1.1.11"

# Description of the issue:
I am testing about 12 devices in the field. 2 of them is not able to receive Cloud to Device messages. All of the devices are running the same OS image, application code with same packages and environment variables. I am able to invoke the direct methods on them, however Cloud to Device messages are not reaching to the device. If you need more info or need to connect to the device let me know and we can arrange it.

Thanks

**Comment by pierreca** [comment]
Hi @bmrpatel 
a couple things we're going to need to debug this: 
1. are all these devices using a different deviceId/key ?
2. please write down the IoT Hub name, device Id, and time at which the problem occurs, it'll be useful later (no need to make this public though if you want to keep it confidential - we'll find a way for you to get that to us)

**Comment by bmrpatel** [comment]
Hello @pierreca 

1. Yes, it is using different deviceId and key.
2. Please let me know how can I communicate this to you?

Thanks

**Comment by pierreca** [comment]
@bmrpatel may be unrelated or not, but I'm working on fixing #41 and I wonder if you could be impacted too: can you try and disable all calls to `onDeviceMethod` in the devices code and let me know if you still have issues receiving C2D messages?

**Comment by pierreca** [comment]
we've released a new package (azure-iot-device@1.1.12) with a fix that may potentially impact you if you were using device methods - can you please give it a test and let us know?

Thanks!

**Comment by pierreca** [comment]
@bmrpatel - still curious to hear if you're still running into issues with the latest version of the SDK that include some fixes that are probably related to what you're seeing.

**Comment by bmrpatel** [comment]
@pierreca It seems to be working so far. Thanks

**Comment by pierreca** [comment]
Good to hear - closing but feel free to reopen if you run into the issue again (or to open another new issue if it looks like a different problem)

**Comment by pk12382** [comment]
Hi All,
I am newbie for IoT, I have implement the sample code given on Azure site to send receive data on Cloud.
I have user RPI and send data to Azure IoT hub and data save in Azure Storage. It working fine.
But I am not able to receive Data from Cloud to Device repeatedly.
I am able to get data from Cloud to Device only once.
How to receive data continue from Cloud to Device.
I have used same example provided on Azure IoT Hub.
